test: add the unit test of parallel diagonalization by ELPA

pxlxingliang opened the pull request:
range: source/src_pdiag

pxlxingliang committed:
test: add the unit test of diagonalization by ELPA
range: source/src_pdiag

caic99 commented on source/src_pdiag/test/diago_elpa_test.cpp:
```diff
@@ -0,0 +1,287 @@
+    int sparsity = 100; // sparsity of H matrix;
+    T *hmatrix = 0;
+    T *smatrix = 0;
+    double *e_lapack = 0; // eigenvalues solvered by LAPACK
```
Typo "solved" here?

caic99 commented on source/src_pdiag/test/diago_elpa_test.cpp:
```diff
@@ -0,0 +1,287 @@
+    int n = 10; // dimension of H/S matrix
+    int nblk = 1; // dimension of sub-matrix
+    int sparsity = 100; // sparsity of H matrix;
+    T *hmatrix = 0;
+    T *smatrix = 0;
+    double *e_lapack = 0; // eigenvalues solvered by LAPACK
+    double *e_elpa = 0;
```
For pointers, please use `nullptr` to initialize.

caic99 replied:
For pointers, please use `nullptr` to initialize.

pxlxingliang committed:
- move function read_hs() to diago_elpa_utils.h
- rename H-KPonts.dat to be H-KPoints.dat
- delete some redundant information output

pxlxingliang committed:
close the print out of redundant information

pxlxingliang committed:
modify S matrix not to be a unitary matrix

pxlxingliang committed:
- test elpa by real Si system produced H/S matrix
- modify the parallel test to be only run one time

pxlxingliang committed:
delete the install of unnecessary file diago_parallel_test.sh
